framework/web/Session.php

```diff
@@ -229,13 +226,29 @@ public function setHasSessionId($value)
     }
 
     /**
-     * Gets the session ID.
-     * This is a wrapper for [PHP session_id()](http://php.net/manual/en/function.session-id.php).
+     * Gets a session ID from the request or null if there is no session ID in the request
+     * @return string|null
+     */
+    private function getRequestSessionIdInternal()
+    {
+        $name = $this->getName();
+        $request = Yii::$app->getRequest();
+        if (ini_get('session.use_cookies') && !empty($_COOKIE[$name])) {
+            return $_COOKIE[$name];
+        }
+        if (!ini_get('session.use_only_cookies') && ini_get('session.use_trans_sid')) {
+            return $request->get($name);
+        }
+        return null;
+    }
+
+    /**
+     * Gets the session ID even the session hasn't yet started
      * @return string the current session ID
      */
     public function getId()
     {
-        return session_id();
+        return session_id() ?: ($this->getRequestSessionIdInternal() ?: '');
```

Comment:
This change is dangerous. If the session is not started, the method may return a requested ID which could be different from the real session ID.